trieste/acquisition/function.py

```diff
@@ -777,24 +778,41 @@ def __init__(
         search_space: SearchSpace,
         num_samples: int = 500,
         penalizer: Callable[..., PenalizationFunction] = None,
+        base_acquisition_function_builder: Optional[SingleModelAcquisitionBuilder] = None,
     ):
         """
         :param search_space: The global search space over which the optimisation is defined.
         :param num_samples: Size of the random sample over which the Lipschitz constant
             is estimated. We recommend scaling this with search space dimension.
+        :param penalizer: The chosen penalization method (defaults to soft penalization).
+        :param base_acquisition_function_builder: Base acquisition function to be
+            penalized (defaults to expected improvement).
```

Comment:
If you are going to restrict this to ExpectedImprovement and MinValueEntropySearch, it should say so in the type (make it `Union[ExpectedImprovement, MinValueEntropySearch]` and there should be a comment about why ("strictly positivie") in the param description.